internal/consensus/types/round_state.go

```diff
@@ -76,14 +77,15 @@ type RoundState struct {
 	// Subjective time when +2/3 precommits for Block at Round were found
 	CommitTime          time.Time           `json:"commit_time"`
 	Validators          *types.ValidatorSet `json:"validators"`
-	Proposal            *types.Proposal     `json:"proposal"`
-	ProposalReceiveTime time.Time           `json:"proposal_receive_time"`
-	ProposalBlock       *types.Block        `json:"proposal_block"`
-	ProposalBlockParts  *types.PartSet      `json:"proposal_block_parts"`
-	LockedRound         int32               `json:"locked_round"`
-	LockedBlock         *types.Block        `json:"locked_block"`
-	LockedBlockParts    *types.PartSet      `json:"locked_block_parts"`
-	Commit              *types.Commit       `json:"commit"`
+	ProposerSelector    selectproposer.ProposerSelector
```

Comment:
why have the proposer selector in the round state? Isn't the selector recoverable from other information?

Reply:
It is, but it requires reading the block store. Here it is an optimization, to not do that lookup in block store every time.

Reply:
okay